pq/reader.go

```diff
@@ -49,39 +49,61 @@ func newReader(accessor *access) (*Reader, error) {
 				pageSize: accessor.PageSize(),
 			},
 		},
-	}, nil
+	}
 }
 
 func (r *Reader) close() {
 	r.active = false
 }
 
 // Available returns the number of unread events that can be read.
-func (r *Reader) Available() uint {
-	if !r.active {
-		return 0
+func (r *Reader) Available() (uint, error) {
+	const op = "pq/reader-available"
+
+	if err := r.canRead(op); err != nil {
+		return 0, err
 	}
 
+	var err reason
 	func() {
-		tx := r.accessor.BeginRead()
-		defer tx.Close()
-		r.updateQueueState(tx)
+		var tx *txfile.Tx
+		tx, err = r.beginTx(op)
+		if err == nil {
+			defer tx.Close()
+			err = r.updateQueueState(op, tx)
+		}
 	}()
+	if err != nil {
+		return 0, err
+	}
 
 	if r.state.cursor.Nil() {
-		return 0
+		return 0, nil
 	}
 
-	return uint(r.state.endID - r.state.id)
+	return uint(r.state.endID - r.state.id), nil
 }
 
 // Begin starts a new read transaction, shared between multiple read calls.
 // User must execute Done, to close the file transaction.
-func (r *Reader) Begin() {
+func (r *Reader) Begin() error {
+	const op = "txfile/reader-begin"
```

Comment:
Why is this prefixed by `txfile` and below the op is prefixed by `pq`?

Reply:
ups. Maybe I should use this pattern:

```
const pkg = "pq/"

func (r *Reader) Begin() error {
  const op = pkg+"reader-begin"
  ...
}
```